### PR TITLE
AC-1917 - (UI) Incorporate valid HTML form elements in the app

### PR DESCRIPTION
**Pull request by shrirupa**
AC-1917 - (UI) Incorporate valid HTML form elements in the app

### What Changed

**On Ab testing page**
- [ ] EditMode.js - Reschedule and Finalize test button should be a submit button inside a `<form>`

**On Domain Pages**
- [ ] In SetupForSending : Re-verify Domain Panel action should be type="submit" instead of having onClick, as they are already inside a `<form>`
- [ ] In SetupBounceDomainSection - Re-verify Domain Panel action should be type="submit" instead of having onClick, as they are already inside a `<form>`
- [ ] In SendingAndBounceDomainSection - Re-verify Domain Panel action should be type="submit" instead of having onClick, as they are already inside a `<form>`
- [ ] In TrackingDnsSection - Re-verify Domain Panel action should be type="submit" instead of having onClick, as they are already inside a `<form>`
- [ ] In VerifyEmailSection | MailboxVerificationModal - should be wrapped in `<form>`

**On Report Builder**
- [ ] MetricsDrawer - this should be wrapped in a `<form>`

**On Snippet**
- [ ] CreatePage and EditPage Save Snippet and Create Snippet should have associated form id and type submit.

**On JoinForm**
- [ ] JoinForm submits on form submit instead of button click.

### How To Test

- onSubmit of these newly added forms work correctly

### To Do

- [ ] Address feedback
- [ ] Check if all form have submit buttons.

### CODE REVIEW

File: src/pages/abTesting/EditMode.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { connect } from 'react-redux';
 import { reduxForm, getFormValues } from 'redux-form';
```
Comment (nstanard) [inline src/pages/abTesting/EditMode.js:4]: General Note: We need to be on the lookout for how our `<Form` instances are used in relation to the submit events. 

For instance:

ManualEntryForm.js doesn't have an `onSubmit` event and I dont' think it will trigger the `handleSubmit` event in our new `src/componnets/Form` component. 

```
...
      <Form
        onKeyDown={this.handleEnter}
        className={styles.DateFields}
        id="datepicker-manualentry-form"
      >
...
```

I think these (date pickers) are the only instance that don't have the `onSubmit` prop, so I think this is the only spot that has the issue but it's something we need to be mindful of and look out for.

Comment (nicklemmon) [reply src/pages/abTesting/EditMode.js:4]: Hey, just a side not - you can submit a form with a `<button>` that is outside of the wrapping `<form/>` using the `form` attribute on that button. Something I learned from Kevin a little while ago.

Comment (shrirupa) [reply src/pages/abTesting/EditMode.js:4]: ok, I checked all the Forms in the app, fortunately there are only six places that have this problem where submit event is not fired - 

src/pages/profile/components/DisableTfaModal.js
src/pages/profile/components/BackupCodesModal.js
src/components/enableTfaForm/EnableTfaForm.js
src/components/datePicker/ManualEntryFormV2.js
src/components/datePicker/ManualEntryForm.js

Comment (shrirupa) [reply src/pages/abTesting/EditMode.js:4]: I am trying to find a way this can be fixed easily in this PR, if not I will cut a separate ticket for this.

Comment (nstanard) [reply src/pages/abTesting/EditMode.js:4]: Since it's more than just the DatePicker component I'd say we cut a new ticket. I'm reviewing this now and we can fix the submit event in the following tickets where we validate each forms fields have their needed attributes.

File: src/pages/join/components/JoinForm.js [hunk 2]
```diff
@@ -70,7 +73,7 @@ export class JoinForm extends Component {
     const pending = loading || submitting || !reCaptchaReady;
 
     return (
-      <Form id="sign-up-form">
+      <Form id="sign-up-form" onSubmit={this.executeRecaptcha}>
```
Comment (nstanard) [inline src/pages/join/components/JoinForm.js:76]: Note to self: Test that recaptcha still works when moving this along.

File: src/pages/domains/components/TrackingDnsSection.js [hunk 3]
```diff
@@ -72,7 +72,7 @@ export default function TrackingDnsSection({ domain, isSectionVisible, title })
                   </Text>
                   <TranslatableText>record for this domain at your DNS provider.</TranslatableText>
                 </p>
-                <Panel.Action onClick={onSubmit}>
+                <Panel.Action type="submit">
```
Comment (nstanard) [inline src/pages/domains/components/TrackingDnsSection.js:75]: I'm not working with a real/valid domain so perhaps that's the issue but I'm not seeing this fire the `<Form` submit event.

File: src/pages/domains/components/SetupBounceDomainSection.js [hunk 4]
```diff
@@ -110,7 +110,7 @@ export default function SetupBounceDomainSection({
                   record for the Hostname and Value for this domain at your DNS provider.
                 </TranslatableText>
               </p>
-              <Panel.Action onClick={onSubmit}>
+              <Panel.Action type="submit" form="verify-bounce-form">
```
Comment (nstanard) [inline src/pages/domains/components/SetupBounceDomainSection.js:113]: We have 4 re-verify buttons in the domains area but I only see three on this PR: 
image

I think we should break this PR into 2 parts. The domain area files can be it's own PR and we can move the rest along. I didn't see anything off with the other files. (and testing/verifying  will be easier if it's broken up into smaller pieces)

Comment (nstanard) [reply src/pages/domains/components/SetupBounceDomainSection.js:113]: I see now that it's already hooked up - and I think that 4th file that's not on the PR is working correctly with the `<Form` submit 👍

File: src/pages/domains/components/SetupForSending.js [hunk 5]
```diff
@@ -106,7 +106,7 @@ export default function SetupForSending({ domain, isSectionVisible }) {
                     record for the Hostname and DKIM value of this domain.
                   </TranslatableText>
                 </p>
-                <Panel.Action onClick={onSubmit}>
+                <Panel.Action type="submit">
```
[the remark this answers is not in the document]
Comment (shrirupa) [reply src/pages/domains/components/SetupForSending.js:109]: I checked the DOM it does get applied.

[the remark this answers is not in the document]
Comment (nstanard) [reply src/pages/domains/components/SetupForSending.js:109]: Was able to verify this just now too 👍

File: src/pages/domains/components/SetupBounceDomainSection.js [hunk 6]
```diff
@@ -110,7 +110,7 @@ export default function SetupBounceDomainSection({
                   record for the Hostname and Value for this domain at your DNS provider.
                 </TranslatableText>
               </p>
-              <Panel.Action onClick={onSubmit}>
```
Comment (nstanard) [inline src/pages/domains/components/SetupBounceDomainSection.js:113]: Verified this location submits the `<Form` submit event. 🟢

File: src/pages/domains/components/VerifyEmailSection.js [hunk 7]
```diff
@@ -71,7 +71,7 @@ export default function VerifyEmailSection({ domain, isSectionVisible }) {
 
 function VerifyButton({ onClick, variant = 'primary', submitting }) {
   return (
-    <Button variant={variant} loading={submitting} onClick={onClick}>
+    <Button variant={variant} loading={submitting} onClick={onClick} type="submit">
```
Comment (nstanard) [inline src/pages/domains/components/VerifyEmailSection.js:74]: This submit event only has a button in the submit event - there is no input being passed through to the `<Form` submit event.

**Comment by shrirupa** [comment]
closing in favor of
https://github.com/SparkPost/2web2ui/pull/2118
https://github.com/SparkPost/2web2ui/pull/2119
